annotations/augmenter.go

```diff
@@ -32,43 +32,50 @@ func (a *annotationAugmenter) AugmentAnnotations(ctx context.Context, depletedAn
 		ctx = tidUtils.TransactionAwareContext(ctx, tid)
 	}
 
-	var conceptIds []string
+	uuids, mappedAnnotations := toMap(canonicalAnnotations)
 
-	for _, ann := range depletedAnnotations {
-		conceptUUID := extractUUID(ann.ConceptId)
-		if conceptUUID != "" {
-			conceptIds = append(conceptIds, conceptUUID)
-		}
-	}
-
-	concepts, err := a.conceptSearchApi.SearchConcepts(ctx, conceptIds)
+	concepts, err := a.conceptSearchApi.SearchConcepts(ctx, uuids)
 
 	if err != nil {
 		log.WithField(tidUtils.TransactionIDKey, tid).
-			WithError(err).Error("Error in augmenting annotation with concept data")
+			WithError(err).Error("Request failed when attempting to augment annotations from UPP concept data")
 		return nil, err
 	}
 
 	var augmentedAnnotations []Annotation
-	for _, ann := range depletedAnnotations {
-		c, found := concepts[ann.ConceptId]
+	for uuid, ann := range mappedAnnotations {
+		concept, found := concepts[uuid]
 		if found {
-			ann.ApiUrl = c.ApiUrl
-			ann.PrefLabel = c.PrefLabel
-			ann.IsFTAuthor = c.IsFTAuthor
-			ann.Type = c.Type
+			ann.ConceptId = concept.ID
+			ann.ApiUrl = concept.ApiUrl
+			ann.PrefLabel = concept.PrefLabel
+			ann.IsFTAuthor = concept.IsFTAuthor
+			ann.Type = concept.Type
 			augmentedAnnotations = append(augmentedAnnotations, ann)
 		} else {
 			log.WithField(tidUtils.TransactionIDKey, tid).
 				WithField("conceptId", ann.ConceptId).
-				Error("Information not found for augmenting concept")
+				Info("Concept data for this annotation was not found, and will be removed from the list of annotations.")
```

Comment:
I wonder whether this should be a `Warn` message? It's not going to prevent things from working, but it is not a generally expected scenario.

Reply:
This is the scenario where the concept for an annotation has been deleted - would this not happen normally? Happy to downgrade to a Warn though, Error seemed far too strong for something we probably wouldn't do anything about.